hibernate-core/src/main/java/org/hibernate/query/sqm/spi/BaseSemanticQueryWalker.java

```diff
@@ -436,7 +436,7 @@ public Object visitValues(SqmValues values) {
 
 	@Override
 	public Object visitWhereClause(SqmWhereClause whereClause) {
-		if ( whereClause == null ) {
+		if ( whereClause == null || whereClause.getPredicate() == null ) {
```

Comment:
IMO instead of doing a null check here again, we should rather make sure that the predicate in the `SqmWhereClause` can never become null by setting the `whereClause` in the `SqmQuerySpec` to `null` instead of setting a `null` `SqmPredicate`. Can you please adapt the set methods instead?

Reply:
Updated the logic to check for `null` and empty arrays when setting multiple predicates.